app/src/main/java/com/android/halanvendordiscovery/domain/vendors/useCase/GetVendorsUseCase.kt

```diff
@@ -0,0 +1,25 @@
+package com.android.halanvendordiscovery.domain.vendors.useCase
+
+import com.android.halanvendordiscovery.domain.vendors.model.Page
+import com.android.halanvendordiscovery.domain.vendors.model.PagedList
+import com.android.halanvendordiscovery.domain.vendors.model.VendorDomainModel
+import com.android.halanvendordiscovery.domain.vendors.repo.VendorRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class GetVendorsUseCase @Inject constructor(
+    private val repository: VendorRepository,
+) {
+
+    suspend operator fun invoke(
+        pageConfig: Page,
+        categoryId: String?,
+        searchKey: String?
+    ): PagedList<VendorDomainModel> {
+        return withContext(Dispatchers.Default) {
```

Comment:
why you need to switch to Dispatchers.Default in the use case?

Reply:
okay , to be clearer , my question is , why do we need to actually switch context of dispatcher within Usecase , is the thread we're using for viewmodel is different or not suitable ?

Reply:
In my ViewModel, I don't explicitly define a specific context. Instead, I launch a coroutine scope where I invoke the useCase. As far as I understand, the coroutine scope automatically switches to the context of the called function.